### PR TITLE
Allow action button to be hidden and provide an alternate place for dismissing the modal

### PR DESCRIPTION
**Pull request by ericjim**
In this PR we do two things:

1) We provide a new prop to the action button, which enables us to create intricate flows within the modal, and only show the action button when a condition is satisfied.

2) We provide an alternate button that dismisses the modal. By default, we place an X on the top-right corner of the inside of the modal. This can be transcluded by the user of the component to replace the X with anything else.

### CODE REVIEW

File: src/layout/modal/modal.tmpl.html [hunk 1]
```diff
@@ -6,13 +6,23 @@
...
+    </div>
+
     <div class="ws-modal-body-container" ng-transclude="body"/>
     <div class="ws-modal-action-container"
       ng-hide="$ctrl.actionInProgress()">
       <a class="button primary-action"
          ng-transclude="action"
          ng-click="$ctrl.doAction()"
-         ng-disabled="$ctrl.invalid">
+         ng-disabled="$ctrl.invalid || $ctrl.disabled"
```
Comment (ericjim) [inline src/layout/modal/modal.tmpl.html]: My intuition tells me that if `disabled` is passed in, the button should also be disabled.

File: src/layout/modal/index.js [hunk 2]
```diff
@@ -8,6 +8,8 @@ const WsModalComponent = {
     disabled: '<',
     secondary: '<',
     invalid: '<',
+    hideAction: '<',
+    alternateDismiss: '<',
```
Comment (ericjim) [inline src/layout/modal/index.js]: Not sure about the name. 🦐

File: src/layout/modal/modal.tmpl.html (hunk 1 above)
Comment (SethDavenport) [reply src/layout/modal/modal.tmpl.html]: :+1:

File: src/layout/modal/index.js (hunk 2 above)
Comment (SethDavenport) [reply src/layout/modal/index.js]: `close`?

Comment (ericjim) [reply src/layout/modal/index.js]: `dismissTop`?

**Pull request merged**